checker/src/main/java/org/checkerframework/checker/regex/RegexTransfer.java

```diff
@@ -79,55 +80,66 @@ public RegexTransfer(CFAbstractAnalysis<CFValue, CFStore, CFTransfer> analysis)
-            // the annotation on that method's definition.)
 
+        // Special cases for some RegexUtil.isRegex and RegexUtil.asRegex methods.  Each one adds an
+        // annotation with elements/arguments if possible, or falls back to just `@Ragex`.
+        // (No special case is needed for isRegex(String) or asRegex(String); their annotations are
+        // sufficient.)
+
+        if (ElementUtils.matchesElement(method, IS_REGEX_METHOD_NAME, String.class, int.class)) {
             CFStore thenStore = result.getRegularStore();
             CFStore elseStore = thenStore.copy();
             ConditionalTransferResult<CFValue, CFStore> newResult =
                     new ConditionalTransferResult<>(result.getResultValue(), thenStore, elseStore);
             JavaExpression firstParam = JavaExpression.fromNode(n.getArgument(0));
 
-            // add annotation with correct group count (if possible,
-            // regex annotation without count otherwise)
             Node count = n.getArgument(1);
             int groupCount;
             if (count instanceof IntegerLiteralNode) {
-                IntegerLiteralNode iln = (IntegerLiteralNode) count;
-                groupCount = iln.getValue();
+                groupCount = ((IntegerLiteralNode) count).getValue();
             } else {
                 groupCount = 0;
             }
-            AnnotationMirror enhancedRegexAnnotation =
-                    factory.createEnhancedRegexAnnotation(
-                            new ArrayList<>(Arrays.asList(0, groupCount)));
+            AnnotationMirror regexNNGroupsAnnotation =
+                    factory.createRegexNNGroupsAnnotation(groupCount, new ArrayList<>());
             if (thenStore.getValue(firstParam) != null) thenStore.clearValue(firstParam);
-            thenStore.insertValue(firstParam, enhancedRegexAnnotation);
+            thenStore.insertValue(firstParam, regexNNGroupsAnnotation);
             return newResult;
         } else if (ElementUtils.matchesElement(
-                method, IS_REGEX_METHOD_NAME, String.class, List.class)) {
-            //            CFStore thenStore = result.getRegularStore();
-            //            CFStore elseStore = thenStore.copy();
-            //            ConditionalTransferResult<CFValue, CFStore> newResult =
-            //                    new ConditionalTransferResult<>(result.getResultValue(),
-            // thenStore, elseStore);
-            //            JavaExpression firstParam = JavaExpression.fromNode(n.getArgument(0));
-            //
-            //            // add annotation with correct group count (if possible,
-            //            // regex annotation without count otherwise)
-            //            Node node = n.getArgument(1);
-            /** TODO Not sure what type of node for Lists. Ask. */
+                method, IS_REGEX_METHOD_NAME, String.class, int.class, int[].class)) {
```

Comment:
Combine this case with the above.  Right now, there is a lot of code duplication, which is confusing to read and is error-prone.  Currently, the branches use different coding paradigms to do the same thing, which makes it hard to determine whether the code is really doing the same thing.